test/e2e/testing-manifests/storage-csi/gce-pd/csi-controller-rbac.yaml

```diff
@@ -107,11 +118,49 @@ metadata:
 subjects:
   - kind: ServiceAccount
     name: csi-gce-pd-controller-sa
+    namespace: default
 roleRef:
   kind: ClusterRole
   name: csi-gce-pd-resizer-role
   apiGroup: rbac.authorization.k8s.io
 
+---
+# xref: https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/rbac-csi-snapshotter.yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: csi-gce-pd-snapshotter-role
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents"]
+    verbs: ["create", "get", "list", "watch", "update", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents/status"]
+    verbs: ["update"]
```

Comment:
also add patch too. I think in general, patch is preferred over update

Reply:
How is patch different from update?  I see in API call, it is usually "update" that gets called.

Reply:
Patch is the preferred way to update api objects. It generally results in fewer failed API updates because you don't need the entire object to be the same if all you want to do is update a single field.  We recently changed external-attacher to use patch instead of update. We should consider changing snapshotter to do the same while we're making breaking changes :-)